hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/SigTransactionMetadata.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.spi.meta;
+
+import com.hedera.node.app.spi.AccountKeyLookup;
+import com.hedera.node.app.spi.key.HederaKey;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Metadata collected when transactions are handled as part of "pre-handle" needed for signature
+ * verification. NOTE : This class may have subclasses in the future.
+ */
+public class SigTransactionMetadata implements TransactionMetadata {
```

Comment:
Can we actually make this a `record`? We won't need to then make the fields protected either, and we can still have the `copy` method (which is nice)

Reply:
We will be needing a sub class for `SigTransactionMetadata` for schedules. So I did not make it a record